packages/core/services/capabilities.js

```diff
@@ -0,0 +1,194 @@
+
+      const capContainerUri = await this.actions.getContainerUri({ webId }, { parentCtx: ctx });
+
+      // Add a reference to the webId document.
+      ctx.call('ldp.resource.patch', {
+        resourceUri: webId,
+        triplesToAdd: [triple(namedNode(webId), namedNode('apods:capabilities'), namedNode(capContainerUri))],
+        webId,
+      });
+
+      this.actions.createInviteCapability({
+        webId,
+        capContainerUri,
+      });
+    },
+  },
+  actions: {
+    createInviteCapability: {
+      params: {
+        webId: { type: 'string', optional: false },
+      },
+      handler: async function (ctx) {
+        const { webId } = ctx.params;
+
+        const capContainerUriPromise = this.actions.getContainerUri({ webId }, { parentCtx: ctx });
+
+        // The profile might not have been created, so we await its URI.
+        const profileUriPromise = (async () => {
+          do {
+            const webIdDoc = await ctx.call('ldp.resource.get', {
+              resourceUri: webId,
+              accept: MIME_TYPES.JSON,
+              webId: 'system',
+            });
+            const profileUri = webIdDoc['url'];
+            if (profileUri) {
+              return profileUri;
+            }
+            delay(500);
+          } while (true);
+        })();
+
+        // The capability container creation might not have finished, so wait until it has.
+        await this.actions.waitForContainerCreation({ containerUri: await capContainerUriPromise }, { parentCtx: ctx });
+
+        // Add an invite capability.
+        const inviteCapUri = await this.actions.post(
+          {
+            containerUri: await capContainerUriPromise,
+            resource: {
+              '@type': 'http://www.w3.org/ns/auth/acl#Authorization',
+              'http://www.w3.org/ns/auth/acl#AccessTo': await profileUriPromise,
+              'http://www.w3.org/ns/auth/acl#Mode': 'http://www.w3.org/ns/auth/acl#Read',
+            },
+            contentType: MIME_TYPES.JSON,
+            webId,
+          },
+          { parentCtx: ctx }
+        );
+        return inviteCapUri;
+      },
+    },
+
+    addCapsContainersWhereMissing: {
+      params: {},
+      handler: async function (ctx) {
+        /** @type {string[]} */
+        const datasets = await ctx.call('pod.list');
+
+        // Add invite cap to each capabilities container where missing.
+        await Promise.all(
+          datasets.map(async (dataset) => {
+            ctx.meta.dataset = dataset;
+
+            const [account] = await ctx.call('auth.account.find', { query: { username: dataset } });
+            const webId = account.webId;
+
+            // TODO: Why is the ldp.resource.get not working (complaining it's not a remote resource)?
+            // const webIdDoc = await ctx.call('ldp.resource.get', {
+            //   resourceUri: webId,
+            //   accept: MIME_TYPES.JSON,
+            //   webId: 'system',
+            // });
```

Comment:
After a while of debugging, it turned out that setting `ctx.meta.dataset` in the `capabilities.addCapsContainersWhereMissing` had some very strange side-effects so that the webId and the dataset/username mismatched. E.g. webid